mayastor/src/bdev/uring_util.rs

```diff
@@ -0,0 +1,87 @@
+use std::{
+    fs::{File, OpenOptions},
+    io::{BufRead, BufReader, ErrorKind},
+    os::unix::fs::OpenOptionsExt,
+};
+
+pub fn fs_supports_direct_io(path: &str) -> bool {
+    // SPDK uring bdev uses IORING_SETUP_IOPOLL which requires O_DIRECT
```

Comment:
nit; what I read is that the FS or Blk dev needs to support polling, not direct IO.

Reply:
According to io_uring_setup(2):
```
       IORING_SETUP_IOPOLL
              Perform  busy-waiting  for an I/O completion, as opposed to get‐
              ting notifications via an asynchronous IRQ (Interrupt  Request).
              The  file  system (if any) and block device must support polling
              in order for this to work.  Busy-waiting provides lower latency,
              but  may  consume  more CPU resources than interrupt driven I/O.
              Currently, this feature is usable  only  on  a  file  descriptor
              opened using the O_DIRECT flag.  When a read or write is submit‐
              ted to a polled context, the application must poll  for  comple‐
              tions  on the CQ ring by calling io_uring_enter(2).  It is ille‐
              gal to mix and match polled and non-polled I/O  on  an  io_uring
              instance.
```
I guess that means both, polling and IO_DIRECT.